pkg/systemd/services/services.jsx

```diff
@@ -252,65 +280,98 @@ class ServicesPageBody extends React.Component {
                     this.listUnits();
                 else
                     this.setState({});
+            } else {
+                debug("visibilitychange to hidden");
             }
         });
 
         const onLocationChanged = () => {
             const options = cockpit.location.options;
+            debug("locationchanged", JSON.stringify(options));
             this.setState({
                 filters: { activeState: JSON.parse(options.activestate || '[]'), fileState: JSON.parse(options.filestate || '[]') },
                 currentTextFilter: decodeURIComponent(options.name || ''),
+                selectedUnits: this.computeSelectedUnits(),
             });
         };
 
         cockpit.addEventListener("locationchanged", onLocationChanged);
```

Comment:
This shouldn't be necessary since `this.props.options` already contains the current options.

Reply:
Hmm, it was necessary before at least, see commit 07a47f514bb74a. But indeed commit 3ea7c472093d6378 seems to have obsoleted this, at least with interactive testing and a local integration test run it works fine now. Nice! I'll add that as a separate commit, I'm getting enough fodder again for a split out PR.

Reply:
Erk, I pressed Enter too soon, sorry; I was meaning to actually wait for the local test run. It does break stuff at least in `testBasic{,Session}` and `testServicesFiltersURLConsistency`. The only thing that seems obsolete is the initial call, the test now passes without that. But I couldn't justify it (without thinking hard about it).

Reply:
> Hmm, it was necessary before at least, see commit [07a47f5](https://github.com/cockpit-project/cockpit/commit/07a47f514bb74a1bcc0995c81ed41e602352eb5d).

That was beause onLocationChanged used `cockpit.location.options`, not `this.props.options`.  In addition to dropping onLocationChanged, the code needs to be changed to compute the filter parameters from `this.props.options`, of course.

Reply:
That said, I see what you mean, having a locationchanged handler here is weird, with also having an `options` property. I'll look at that next after sifting through the other flakes.